### PR TITLE
merge addon local repo

### PR DESCRIPTION
**Pull request by ryanzhang-oss**
Merge addon local repo

- Added a helm chart
- Added the core controllers to the pkg 
- Clean up the pkg directory so that it aligns well with our current structure
- Combined examples
- Fix various issues
- Fix Make CI e2e test work

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -45,4 +45,5 @@ jobs:
           version: "v0.7.0"
 
       - name: Run e2e tests
-        run: make go-integration
+        run: make e2e-test
```
Comment (zzxwill) [inline .github/workflows/go.yml:48]: With this renaming, should we change the name of this directory `https://github.com/crossplane/oam-kubernetes-runtime/tree/master/test/integration` ?

File: .gitignore [hunk 2]
```diff
@@ -29,7 +29,6 @@ vendor/
 .DS_Store
 bin/
 tmp/
-crds/
```
Comment (zzxwill) [inline .gitignore:32]: If we move this line, developers might apply the outdated manifests if the forget to `make generate` and use those manifests from this present crds folder, I am afraid.

File: README.md [hunk 3]
```diff
@@ -14,12 +14,97 @@ In addition, we created this library with the following goals in mind
...
  Kubernetes platform builders.
 
-## Getting started
-Check out [DEVELOPMENT.md](./DEVELOPMENT.md) to see how to develop with OAM Kubernetes runtime
+
+## Prerequisites
+
+- Kubernetes v1.16+
+- Helm 3
+- [Crossplane](https://github.com/crossplane/crossplane) v0.11+ installed
```
Comment (zzxwill) [inline README.md:22]: A must-have?

File: README.md [hunk 4]
```diff
@@ -14,12 +14,97 @@ In addition, we created this library with the following goals in mind
...
+
+## Install OAM controllers
+
+#### Clone this repo
+
+```console
+git clone git@github.com:crossplane/oam-kubernetes-runtime.git
+cd ./oam-kubernetes-runtime
```
Comment (wonderflow) [inline README.md:30]: We can make it to be helm install without git clone in following PR.

File: README.md [hunk 5]
```diff
@@ -14,12 +14,97 @@ In addition, we created this library with the following goals in mind
...
+```
+
+#### Install controllers
+
+```console
+kubectl create namespace oam-system
+helm install controller -n oam-system ./charts/oam-core/
```
Comment (wonderflow) [inline README.md:37]:
```suggestion
helm install oam-runtime -n oam-system ./charts/oam-core/ 
```

File: README.md [hunk 6]
```diff
@@ -14,12 +14,97 @@ In addition, we created this library with the following goals in mind
...
+And a service looking like below
+```console
+kubectl get services
+AME                                             TYPE       CLUSTER-IP     EXTERNAL-IP   PORT(S)    AGE
+example-appconfig-workload-deployment-service   NodePort   10.96.78.215   <none>        8080/TCP   28s
+```
+
+## Cleanup
+```console
+helm uninstall controller -n oam-system
```
Comment (wonderflow) [inline README.md:96]:
```suggestion
helm uninstall oam-runtime -n oam-system
```

File: main.go [hunk 7]
```diff
@@ -37,12 +37,13 @@ func main() {
 		o.Development = true
 	}))
 
-	oamLog := ctrl.Log.WithName("oam-runtime")
+	oamLog := ctrl.Log.WithName("oam controller")
```
Comment (wonderflow) [inline main.go:40]: oam-runtime is a good name, why change it?

File: Makefile [hunk 8]
```diff
@@ -96,4 +97,28 @@ oam-runtime.help:
 
 help-special: oam-runtime.help
 
-.PHONY: oam-runtime.help help-special
\ No newline at end of file
+.PHONY: oam-runtime.help help-special
```
[the remark this answers is not in the document]
Comment (ryanzhang-oss) [reply Makefile:100]: What does this do?

[the remark this answers is not in the document]
Comment (wonderflow) [reply Makefile:100]: https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html

**Pull request merged**